### PR TITLE
Users can leave events and invite others users if appropriate

### PR DESCRIPTION
**Pull request by VanAnderson**
This combines a couple different stories. Users can now leave an event from the event show page just like they can join. Per comments from @jah2488 logic has been stripped from the views and put into helpers.

Users can also invite other users if they are going to an event and the event has an open invite(or is public).

### CODE REVIEW

File: app/controllers/invites_controller.rb [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
 
   def update
     get_invite
+<<<<<<< HEAD
+    @event = @invite.event
+    @room = @event.room
+    @campus = @event.campus
+    @invite.assign_attributes(invite_params.merge(user_id: current_user.id)) if @invite.user_id == current_user.id
+    save_for_html_json(@invite, "/events/show") { :back }
+=======
```
Comment (reicht) [inline app/controllers/invites_controller.rb]: Is this some merge conflict formatting that is hanging out here?

Comment (VanAnderson) [inline app/controllers/invites_controller.rb]: Yikes!

Comment (VanAnderson) [inline app/controllers/invites_controller.rb]: I didn't save all before the commit, should be good now.

**Comment by jaredpowers** [comment]
👍

**Comment by brentmartin** [comment]
👍

**Pull request merged**